stefaniaPopescu26-2 - Submission Assignment 2

stefaniaPopescu26 opened the pull request.

kkasiacodeu commented on stefaniaPopescu26/assignment2/CommonAncestor.java:
```diff
@@ -0,0 +1,227 @@
+            if (left != null)
+                LCAleft = LCA(left, data1, data2);
+
+            if (right != null)
```
These calls might not be necessary if the LCAleft is not null. You can move lines 186-187 before this to return early.

stefaniaPopescu26 commented:
I made the changes. Thank you for pointing out everything that was unnecessary. I added the tests. It's the first time that I use junit, so I can't wait to read your feedback :)

kkasiacodeu commented on stefaniaPopescu26/assignment2/Tree.java:
```diff
@@ -0,0 +1,91 @@
+public class Tree<T> {
+    private Node<T> root;
+    private String inorder;
```
This doesn't have to be a global variable. You only need it in the inorderTraversal method. Global variables should only be added if they hold a state that needs to be accessed at different times and from different methods (like root in this class). But the inorder can be locally computed in the inorderTraversal and returned as a return value there.

kkasiacodeu commented on stefaniaPopescu26/assignment2/Tree.java:
```diff
@@ -0,0 +1,91 @@
+        }
+    }
+
+    public void inorderTraversal(Node<T> node) {
```
make this method private since it is only a helper method for getInorder() and is never used from the outside of this class.

kkasiacodeu commented on stefaniaPopescu26/assignment2/TreeTest.java:
```diff
@@ -0,0 +1,51 @@
+    }
+
+    @Test
+    public void testAdd1() {
```
This test has all the test cases that the other test have too, there is no need for this repetition :)

kkasiacodeu commented on stefaniaPopescu26/assignment2/TreeTest.java:
```diff
@@ -0,0 +1,51 @@
+
+public class TreeTest {
+
+    public Tree<Integer> getTestTree1() {
```
Try making namings of your methods more meaningful: for example (and this is just my suggestion here, feel free to go with your own idea):
getTestTree1 -> getTestTreeInts
getTestTree2 -> getTestTreeChars
testAdd1 -> testAddInts
testAdd2 - >testAddChars
testAdd3 -> testInOrederNull

kkasiacodeu commented on stefaniaPopescu26/assignment2/Ancestors.java:
```diff
@@ -0,0 +1,63 @@
+        Vector<Node<T>> ancestors = new Vector<>();
+        Node<T> node = findNode(getRoot(), data);
+
+        if (node == null) {
```
As I already wrote in the previous comment: this will make calling getAncestors with a data that is not even in the tree return an empty list. How is that distinguishable from calling getAncestors on the root Node? I think it would be good if the algorithm behaved differently for those two cases.

kkasiacodeu commented on stefaniaPopescu26/assignment2/AncestorsTest.java:
```diff
@@ -0,0 +1,54 @@
+
+public class AncestorsTest {
+
+    public Ancestors<Integer> getTree() {
```
As per comment in the tests for the Tree class: can you try to name the test cases in a way that would reflect what is being tested? E.g. testGetAncestors_invalidNode. testGetAncestors_rootNode, etc.

stefaniaPopescu26 commented:
Sorry for making the changes so late. I had to study harder for my last exam. I renamed the methods in all the test files and solved the problem from the previous review so that the code will return different messages for those two cases. I didn't understand correctly what you told me last time, I'm sorry. I've also removed the global variable from Tree.java.

stefaniaPopescu26 commented:
Thank you :)
